lib/libc/minimal/CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
+# As minimal libc will be built as a zephyr_library, clear c_library to
+# prevent accidentally requiring or linking against another libc implementation.
+set_property(TARGET linker PROPERTY c_library "")
```

Comment:
One other idea was to modify `link_order_library` in the various `linker_libraries.cmake` files to omit `c` if we're building with `CONFIG_MINIMAL_LIBC`. That did seem to work (and seems analogous to what was done for `CONFIG_MINIMAL_LIBCPP`), but this was a smaller change (and is somewhat similar to what is done for picolibc-as-a-module) so I ended up doing this instead--not sure if one is better than the other otherwise or if there is something entirely different that would be preferred!

Reply:
You probably need to also change `math_library` and `c++_library` to make sure we don't mix other incompatible bits. This way we'll get a build error rather than a runtime error.

Reply:
> You probably need to also change math_library and c++_library to make sure we don't mix other incompatible bits.

actually, one intention in the new design is that you don't have to change all the definitions of the libraries, if all you want is to change the libs used and/or the order in which they are linked.
In such cases, the `link_order_library` property should be adjusted instead.

As the minimal libc is inside the whole-archive flags and it appears we don't want other bits, then I would suggest to do:
```
set_property(TARGET linker PROPERTY link_order_library "")
```

A future option is to change the minimal libc into a named library, and then link it specifically at the end, similar to what is done for picolibc, but that is a little more work, and thus I don't request it for this PR.